kvdb/sqlbase/readwrite_tx.go

```diff
@@ -49,7 +49,8 @@ func newReadWriteTx(db *db, readOnly bool) (*readWriteTx, error) {
 	tx, err := db.db.BeginTx(
 		context.Background(),
 		&sql.TxOptions{
-			ReadOnly: readOnly,
+			ReadOnly:  readOnly,
+			Isolation: sql.LevelSerializable,
```

Comment:
Just FYI from the [sqlite docs](https://www.sqlite.org/isolation.html),
> Transactions in SQLite are SERIALIZABLE.

Also think this will create some overhead, but I guess we don't need to worry about it now as there are many other ways to improve performance.

Reply:
Yeah this doesn't do anything for `sqlite`, only for `postgres`. 

Correct that it adds some over head, but that's the tradeoff for safety when it comes to db concurrency.